import complex.h from c10

manuelcandales opened the pull request:
Summary: Import complex type and utils from c10 in order to do arithmetic with complex numbers in ExecuTorch kernels.

Reviewed By: digantdesai

Differential Revision: D72197508

swolchok reviewed (approved):
seems legit as long as the c10 headers are exact copies

I thought this would be bc-breaking; didn't realize that real_ and imag_ on c10::complex are public.

manuelcandales committed:
import complex.h from c10 (#10155)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/10155

Import complex type and utils from c10 in order to do arithmetic with complex numbers in ExecuTorch kernels.

Reviewed By: digantdesai, swolchok

Differential Revision: D72197508
